modules/ting_search/ting_search.module

```diff
@@ -445,20 +449,23 @@ function ting_search_search_execute($keys = NULL, $conditions = NULL) {
         $results_per_page = $conditions['size'];
       }
 
-      // TODO BBS-SAL: port, currently ignored.
       // Set sort order.
       if (!empty($conditions['sort'])) {
-        $options['sort'] = $conditions['sort'];
+        // Get all sort-options available from the provider.
+        $available_sort_options = ting_search_sort_options();
+
+        // Determine which sort-option the user has specified, pick out the
+        // TingSearchSort instance specified by the provider and add it to the
+        // query.
+        if (isset($available_sort_options[$conditions['sort']]) && !empty($available_sort_options[$conditions['sort']]['sort'])) {
```

Comment:
Why do you need the isset here? `empty()` should return `false` without raising warnings etc.